### PR TITLE
Added additional tests to cover HexaryTrie MPT proofs.

### PR DESCRIPTION
**Pull request by web3-developer**

### CODE REVIEW

File: tests/trie/test_hexary_proof.nim [hunk 1]
```diff
@@ -59,3 +60,88 @@ suite "MPT trie proof verification":
...
+    let db = newMemoryDB()
+    var trie = initHexaryTrie(db)
+
+    let 
+      proof = trie.getBranch("not-exist".toBytes)
+      res = verifyMptProof(proof, trie.rootHash, "not-exist".toBytes, "not-exist".toBytes)
+
+    check: 
+      trie.rootHash == keccakHash(emptyRlp)
+      proof.len() == 1 # Maybe this should return an empty list?
```
Comment (kdeme) [inline tests/trie/test_hexary_proof.nim:77]: I think it makes sense that it is `len() == 1` considering it drops in the empty rlp value?

The fact that the result is `InvalidProof` is however inconsistent for when you would add a "not-exist` into a not empty trie. The emptry trie thing is quite the edge case though.

File: tests/trie/test_hexary_proof.nim [hunk 2]
```diff
@@ -59,3 +60,88 @@ suite "MPT trie proof verification":
...
       res.isMissing()
+
+  # proof test ref:
+  # - https://github.com/ethereum/go-ethereum/blob/master/trie/proof_test.go
+  # - https://github.com/ethereum/py-trie/blob/master/tests/test_proof.py
+  test "Validate proof for empty trie":
+    let db = newMemoryDB()
+    var trie = initHexaryTrie(db)
+
+    let
```
Comment (kdeme) [inline tests/trie/test_hexary_proof.nim:71]: I noticed quite some trailing whitespace in this and previous PR. You might want to setup up some marking and/or trimming of that.

Comment (web3-developer) [reply tests/trie/test_hexary_proof.nim:71]: Do you mean at the bottom of the file? Or spaces after the let?

File: tests/trie/test_hexary_proof.nim (hunk 1 above)
Comment (web3-developer) [reply tests/trie/test_hexary_proof.nim:77]: Yeah it is an edge case so probably not worth spending too much time on. I raised it because I noticed that this library behaves differently than the Rust implementation which returns an empty proof when the trie is empty.

File: tests/trie/test_hexary_proof.nim (hunk 2 above)
Comment (kdeme) [reply tests/trie/test_hexary_proof.nim:71]: spaces after the let and some other locations.

File: tests/trie/test_hexary_proof.nim (hunk 1 above)
Comment (kdeme) [reply tests/trie/test_hexary_proof.nim:77]: I'd suggest to add a Note in comment that mentions this and perhaps also link to this rust library.

Comment (web3-developer) [reply tests/trie/test_hexary_proof.nim:77]: Sure, will do

File: tests/trie/test_hexary_proof.nim (hunk 2 above)
Comment (web3-developer) [reply tests/trie/test_hexary_proof.nim:71]: Ok thanks

**Comment by web3-developer** [comment]
@kdeme I've updated the code as you suggested. Ready for the final review. Thanks

**Review by kdeme — APPROVED** [review]
